src/drive/web/modules/views/Trash/useActions.jsx

```diff
@@ -0,0 +1,48 @@
+import React, { useContext } from 'react'
+import { SharingContext } from 'cozy-sharing'
+
+import { useClient } from 'cozy-client'
+import { ModalContext } from 'drive/lib/ModalContext'
+import keyBy from 'lodash/keyBy'
+
+import DestroyConfirm from 'drive/web/modules/trash/components/DestroyConfirm'
+
+const useActions = () => {
+  const { pushModal, popModal } = useContext(ModalContext)
+  const { refresh } = useContext(SharingContext)
+
+  const client = useClient()
+
+  const actions = {
+    restore: {
+      icon: 'restore',
+      action: async files => {
+        for (const file of files) {
+          await client.collection('io.cozy.files').restore(file.id)
+        }
+        refresh()
+      }
+    },
+    destroy: {
+      icon: 'destroy',
+      action: files =>
+        pushModal(
+          <DestroyConfirm
+            fileCount={files.length}
+            confirm={async () => {
+              for (const file of files) {
+                await client
+                  .collection('io.cozy.files')
+                  .deleteFilePermanently(file.id)
+              }
+            }}
+            onClose={popModal}
+          />
```

Comment:
I wonder why the confirm function needs to be passed here. I think DestroyConfirm could have a `files` prop, and could provide the `confirm` implementation to downstream components (instead of having the confirm implementation here. This way it would be testable on its own, and would not need to be wired up through `useActions`. And the fileCount would be taken directly from the `files` property.